kolibri/plugins/learn/assets/src/views/ExamPage/AnswerHistory.vue

```diff
@@ -32,29 +32,62 @@
 
   import { mapState } from 'vuex';
 
+  function isAboveContainer(element, container) {
+    return element.offsetTop < container.scrollTop;
+  }
+
+  function isBelowContainer(element, container) {
+    return element.offsetTop + element.offsetHeight > container.offsetHeight + container.scrollTop;
+  }
+
   export default {
     name: 'AnswerHistory',
     props: {
       questionNumber: {
         type: Number,
         required: true,
       },
+      // hack to get access to the scrolling pane
+      wrapperComponentRefs: {
+        type: Object,
+        required: true,
+      },
     },
     computed: {
       ...mapState('examViewer', ['questions']),
       ...mapState({ attemptLogs: 'examAttemptLogs' }),
     },
+    watch: {
+      questionNumber(index) {
+        // If possible, scroll it into view
+        const element = this.$refs[`item-${index}`][0];
+        if (element && element.scrollIntoView && this.wrapperComponentRefs.questionListWrapper) {
+          const container = this.wrapperComponentRefs.questionListWrapper.$el;
+          if (isAboveContainer(element, container)) {
+            element.scrollIntoView({ block: 'start', inline: 'nearest', behavior: 'smooth' });
+          } else if (isBelowContainer(element, container)) {
+            element.scrollIntoView({ block: 'end', inline: 'nearest', behavior: 'smooth' });
+          }
+        }
+      },
```

Comment:
I think an alternate approach could be to emit another event for something like `questionRendered` (or perhaps just a `$nextTick` on the existing event) and then expose public methods of `isAboveContainer(container)`, `isBelowContainer(container)`, and `scrollIntoView(opts)`. It still would seem a little awkward because this component has a list of questions, but considering it accepts a `questionNumber` prop, I think exposing those methods which make decisions based on that prop could make sense.